src/components/ChapterForm.tsx

```diff
@@ -0,0 +1,67 @@
+"use client";
+import AutoForm from "@/components/ui/auto-form";
+import * as z from "zod";
+import { Card, CardContent, CardHeader, CardTitle } from "./ui/card";
+import { Button } from "@/components/ui/button";
+import { api } from "~/utils/api";
+
+import React, { useState } from "react";
+const chapterInputs = z.object({
+  title: z
+    .string({
+      required_error: "Title is required.",
+    })
+    .min(2, {
+      message: "Title must be at least 2 characters.",
+    }),
+});
+
+interface ChapterFormProps {
+  bookId: string;
+}
+
+const ChapterForm = ({ bookId }: ChapterFormProps): JSX.Element => {
+  const { mutate, data, isSuccess } = api.book.addChapter.useMutation({
```

Comment:
do we need `data` here?

Reply:
as best practice, we should make sure to name our vars, which doesn't really matter for components this small but for larger components keeps everything in check  
```suggestion
  const { mutate: addChapterMutation, data, isSuccess: isAddChapterMutationSuccessful } = api.book.addChapter.useMutation({
```